tests/write_operations_test.go

```diff
@@ -0,0 +1,813 @@
+		s.Require().NoError(err1)
+		
+		err2 := tx2.ReplaceDataFiles(s.ctx, originalFiles, []string{consolidatedPath2}, nil)
+		s.Require().NoError(err2)
+		
+		// First should succeed
+		_, err1 = tx1.Commit(s.ctx)
+		s.Require().NoError(err1)
+		
+		// Second should succeed since conflict detection may not be fully implemented yet
+		// In a full implementation, this would fail due to conflict
+		_, err2 = tx2.Commit(s.ctx)
+		if err2 != nil {
+			s.T().Logf("Transaction conflict detected: %v", err2)
+		} else {
+			s.T().Log("Transaction completed without conflict - conflict detection may not be fully implemented")
+		}
+	})
+}
+
+func (s *WriteOperationsTestSuite) TestOverwriteFiles() {
+	s.Run("OverwriteByPartition", func() {
+		// Note: Partition-specific operations require more complex setup
+		// For now, demonstrate basic overwrite capability
+		ident := table.Identifier{"default", "overwrite_partition_test"}
+		tbl, originalFiles := s.createTableWithData(ident, 1)
+		
+		initialSnapshot := tbl.CurrentSnapshot()
+		
+		// Create new data to simulate partition overwrite
+		mem := memory.DefaultAllocator
+		newData, err := array.TableFromJSON(mem, s.arrSchema, []string{
+			`[
+				{"id": 10, "data": "new_foo", "ts": 1672531200000000},
+				{"id": 20, "data": "new_bar", "ts": 1672534800000000}
+			]`,
+		})
+		s.Require().NoError(err)
+		defer newData.Release()
+		
+		// Write new file for overwrite
+		newFilePath := fmt.Sprintf("%s/data/new_partition_data.parquet", s.location)
+		fs := s.getFS(tbl)
+		s.writeParquet(fs, newFilePath, newData)
+		
+		// Overwrite with new data
+		tx := tbl.NewTransaction()
+		err = tx.ReplaceDataFiles(s.ctx, originalFiles, []string{newFilePath}, nil)
+		s.Require().NoError(err)
+		
+		newTbl, err := tx.Commit(s.ctx)
+		s.Require().NoError(err)
+		
+		// Verify new snapshot
+		newSnapshot := newTbl.CurrentSnapshot()
+		s.Require().NotNil(newSnapshot)
+		s.NotEqual(initialSnapshot.SnapshotID, newSnapshot.SnapshotID)
+	})
+	
+	s.Run("OverwriteWithFilter", func() {
+		// Test overwrite with row-level filters
+		ident := table.Identifier{"default", "overwrite_filter_test"}
+		tbl, originalFiles := s.createTableWithData(ident, 1)
+		
+		// Create replacement data
+		mem := memory.DefaultAllocator
+		replacementData, err := array.TableFromJSON(mem, s.arrSchema, []string{
+			`[
+				{"id": 1, "data": "updated_foo", "ts": 1672531200000000},
+				{"id": 4, "data": "new_data", "ts": 1672549200000000}
+			]`,
+		})
+		s.Require().NoError(err)
+		defer replacementData.Release()
+		
+		// For now, this demonstrates file-level replacement
+		// True row-level filtering would require delete files
+		consolidatedPath := fmt.Sprintf("%s/data/replacement.parquet", s.location)
+		fs := s.getFS(tbl)
+		s.writeParquet(fs, consolidatedPath, replacementData)
+		
+		// For this test, we'll demonstrate overwrite by creating a new file
+		// and replacing specific known files rather than discovering them dynamically
```

Comment:
need an assertion of this row

Reply:
added https://github.com/apache/iceberg-go/pull/497/commits/3500678da94bc7b2f0eb91c2523cec5fdefd4cf9